index.bs

```diff
@@ -74,903 +60,162 @@ specification uses that specification and terminology.
-        from.
-     1. Let |abortInfo| be the first argument.
-     1. Start the closing procedure by sending a message to the remote side
-        indicating that the stream has been aborted (for QUIC, this is a
-        STOP_SENDING frame) with its error code set to the value of
-        |abortInfo.errorCode|.
-
-: <dfn method for="IncomingStream">arrayBuffer()</dfn>
-:: A convenience method that asynchronously reads all the contents of
-   |IncomingStream.readable| and returns it as an ArrayBuffer.  This locks the
-   stream while reading, just as if |IncomingStream.readable.getReader()| were
-   used.
-
-# Interface `WebTransportStream` #  {#web-transport-stream}
-
-A collection of common attributes and methods of all streams.
+The {{CompressionStream}}(format) constructor, when invoked, must run these steps:
+    1. If <dfn for=CompressionStream>format</dfn> is unsupported in CompressionStream, then throw a TypeError.
+    1. Let *cs* be a new CompressionStream object.
+    1. Set *cs*'s *format* to <a for=CompressionStream>format</a>.
+    1. Let *startAlgorithm* be an algorithm that takes no arguments and returns nothing.
+    1. Let *transformAlgorithm* be an algorithm which takes a *chunk* argument and runs the <a>compress and enqueue a chunk</a> algorithm with *cs* and *chunk*.
+    1. Let *flushAlgorithm* be an algorithm which takes no argument and runs the <a for=CompressionStream>flush and enqueue</a> algorithm with *cs*.
+    1. Let *transform* be the result of calling <a abstract-op>CreateTransformStream</a>(*startAlgorithm*, *transformAlgorithm*, *flushAlgorithm*).
+    1. Set *cs*'s <a for=CompressionStream>transform</a> to *transform*.
+    1. Return *cs*.
 
-<pre class="idl">
-[ Exposed=Window ]
-interface WebTransportStream {
-    readonly attribute unsigned long long streamId;
-    readonly attribute WebTransport transport;
-};
-</pre>
-
-## Attributes ##  {#web-transport-stream-attributes}
-
-: <dfn for="WebTransportStream" attribute>streamId</dfn>
-:: The readonly attribute referring to the ID of the {{WebTransportStream}}
-   object.
-: <dfn for="WebTransportStream" attribute>transport</dfn>
-:: The readonly attribute referring to the related {{WebTransport}} object.
-
-# Interface `BidirectionalStream` #  {#bidirectional-stream}
-
-<pre class="idl">
-[ Exposed=Window ]
-interface BidirectionalStream : WebTransportStream {
-};
-BidirectionalStream includes OutgoingStream;
-BidirectionalStream includes IncomingStream;
-</pre>
+The <dfn>compress and enqueue a chunk</dfn> algorithm, given a CompressionStream object *cs* and a *chunk*, runs these steps:
+    1. If *chunk* is not a {{BufferSize}} type, then throw a TypeError.
+    1. Let *buffer* be the result of compressing *chunk* with *cs*'s *format*. If this throws an exception, then return a promise rejected with that exception.
+    1. Let *controller* be *cs*'s transform.[[TransformStreamController]].
```

Comment:
Please put a backslash before `[[TransformStreamController]]`. This will stop the Bikeshed tool from trying to link it. The backslash will be correctly removed from the output.

Also do the same for `[[readable]]` and `[[writable]]`.